### PR TITLE
V3 async transfer

### PR DESCRIPTION
**Pull request by jeremeamia**
@mtdowling Opening this so you can preview the changes. Still need to do the tests.

### CODE REVIEW

File: src/S3/Transfer.php [hunk 1]
```diff
@@ -2,7 +2,12 @@
 namespace Aws\S3;
 
 use Aws;
-use Aws\Command;
+use Aws\CommandInterface;
+use Aws\ResultInterface;
+use GuzzleHttp\Promise;
+use GuzzleHttp\Psr7;
+use GuzzleHttp\Promise\PromisorInterface;
+use Symfony\Component\Yaml\Exception\RuntimeException;
```
Comment (mtdowling) [inline src/S3/Transfer.php]: Whoops!

**Comment by mtdowling** [comment]
This looks awesome.

**Comment by jeremeamia** [comment]
Merged manually.

**Comment by mtdowling** [comment]
:tada: :balloon:

**Pull request closed without merging**